src/tooling/docs-assembler/Sourcing/RepositorySourcesFetcher.cs

```diff
@@ -91,8 +102,23 @@ await Task.Run(() =>
 					checkouts.Add(RepositorySourcer.CloneRef(repo.Value, gitRef, fetchLatest));
 				}, c);
 			}).ConfigureAwait(false);
-		return checkouts;
+		await File.WriteAllTextAsync(
+			Path.Combine(context.CheckoutDirectory.FullName, CheckoutResult.LinkRegistrySnapshotFileName),
+			LinkRegistry.Serialize(linkRegistry),
+			ctx
+		);
+		return new CheckoutResult
+		{
+			Checkouts = checkouts,
+			LinkRegistrySnapshot = linkRegistry
+		};
 	}
+
+	public async Task WriteLinkRegistrySnapshot(LinkRegistry linkRegistrySnapshot, Cancel ctx = default) => await File.WriteAllTextAsync(
```

Comment:
```suggestion
	public async Task WriteLinkRegistrySnapshot(LinkRegistry linkRegistrySnapshot, Cancel ctx = default) => await context.WriteFileSystem.File.WriteAllTextAsync(
```

Reply:
Also changed it here: https://github.com/elastic/docs-builder/pull/1316/commits/6c2f39d8b6410682fc8988bce98e23bcc6f38b49